// @coinbase/cdp-sdk/_cjs/openapi-client/generated/coinbaseDeveloperPlatformAPIs.schemas.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SplValueCriterionOperator = exports.SplValueCriterionType = exports.SplAddressCriterionOperator = exports.SplAddressCriterionType = exports.SolValueCriterionOperator = exports.SolValueCriterionType = exports.SolAddressCriterionOperator = exports.SolAddressCriterionType = exports.SignEvmTypedDataRuleOperation = exports.SignEvmTypedDataRuleAction = exports.SignEvmTypedDataVerifyingContractCriterionOperator = exports.SignEvmTypedDataVerifyingContractCriterionType = exports.SignEvmTypedDataFieldCriterionType = exports.EvmTypedNumericalConditionOperator = exports.EvmTypedAddressConditionOperator = exports.SignEvmMessageRuleOperation = exports.SignEvmMessageRuleAction = exports.EvmMessageCriterionType = exports.SendEvmTransactionRuleOperation = exports.SendEvmTransactionRuleAction = exports.EvmNetworkCriterionOperator = exports.EvmNetworkCriterionNetworksItem = exports.EvmNetworkCriterionType = exports.SignEvmTransactionRuleOperation = exports.SignEvmTransactionRuleAction = exports.NetUSDChangeCriterionOperator = exports.NetUSDChangeCriterionType = exports.EvmDataCriterionType = exports.EvmDataParameterConditionListOperator = exports.EvmDataParameterConditionOperator = exports.AbiInputType = exports.AbiFunctionType = exports.AbiStateMutability = exports.KnownAbiType = exports.EvmAddressCriterionOperator = exports.EvmAddressCriterionType = exports.EthValueCriterionOperator = exports.EthValueCriterionType = exports.ListEvmTokenBalancesNetwork = exports.EvmSwapsNetwork = exports.SpendPermissionNetwork = exports.EvmUserOperationStatus = exports.EvmUserOperationNetwork = exports.EvmEip7702DelegationOperationStatus = exports.EvmEip7702DelegationNetwork = exports.ErrorType = exports.OAuth2ProviderType = exports.DeveloperJWTAuthenticationType = exports.SmsAuthenticationType = exports.EmailAuthenticationType = void 0;
exports.X402SupportedPaymentKindScheme = exports.X402SettleErrorReason = exports.X402VerifyInvalidReason = exports.X402V1PaymentRequirementsNetwork = exports.X402V1PaymentRequirementsScheme = exports.X402V2PaymentRequirementsScheme = exports.X402V1PaymentPayloadNetwork = exports.X402V1PaymentPayloadScheme = exports.X402Version = exports.OnchainDataResultSchemaColumnsItemType = exports.ListSolanaTokenBalancesNetwork = exports.PolicyScope = exports.SignEndUserSolMessageRuleOperation = exports.SignEndUserSolMessageRuleAction = exports.SendEndUserSolTransactionRuleOperation = exports.SendEndUserSolTransactionRuleAction = exports.SignEndUserSolTransactionRuleOperation = exports.SignEndUserSolTransactionRuleAction = exports.SignEndUserEvmTypedDataRuleOperation = exports.SignEndUserEvmTypedDataRuleAction = exports.SignEndUserEvmMessageRuleOperation = exports.SignEndUserEvmMessageRuleAction = exports.SendEndUserEvmTransactionRuleOperation = exports.SendEndUserEvmTransactionRuleAction = exports.SignEndUserEvmTransactionRuleOperation = exports.SignEndUserEvmTransactionRuleAction = exports.SendUserOperationRuleOperation = exports.SendUserOperationRuleAction = exports.PrepareUserOperationRuleOperation = exports.PrepareUserOperationRuleAction = exports.SignEvmHashRuleOperation = exports.SignEvmHashRuleAction = exports.SignSolMessageRuleOperation = exports.SignSolMessageRuleAction = exports.SolMessageCriterionType = exports.SendSolTransactionRuleOperation = exports.SendSolTransactionRuleAction = exports.SolNetworkCriterionOperator = exports.SolNetworkCriterionNetworksItem = exports.SolNetworkCriterionType = exports.SignSolTransactionRuleOperation = exports.SignSolTransactionRuleAction = exports.ProgramIdCriterionOperator = exports.ProgramIdCriterionType = exports.SolDataCriterionType = exports.SolDataParameterConditionListOperator = exports.SolDataParameterConditionOperator = exports.KnownIdlType = exports.MintAddressCriterionOperator = exports.MintAddressCriterionType = void 0;
exports.RequestSolanaFaucetBodyToken = exports.SendSolanaTransactionBodyNetwork = exports.CreatePolicyBodyScope = exports.ListPoliciesScope = exports.RequestEvmFaucetBodyToken = exports.RequestEvmFaucetBodyNetwork = exports.SendEvmTransactionBodyNetwork = exports.ImportEndUserBodyKeyType = exports.ListEndUsersSortItem = exports.OnrampLimitType = exports.OnrampUserIdType = exports.OnrampQuotePaymentMethodTypeId = exports.OnrampPaymentLinkType = exports.OnrampOrderStatus = exports.OnrampOrderFeeType = exports.OnrampOrderPaymentMethodTypeId = exports.X402SupportedPaymentKindNetwork = void 0;
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EmailAuthenticationType = {
    email: "email",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SmsAuthenticationType = {
    sms: "sms",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.DeveloperJWTAuthenticationType = {
    jwt: "jwt",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OAuth2ProviderType = {
    google: "google",
    apple: "apple",
    x: "x",
    telegram: "telegram",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ErrorType = {
    already_exists: "already_exists",
    bad_gateway: "bad_gateway",
    client_closed_request: "client_closed_request",
    faucet_limit_exceeded: "faucet_limit_exceeded",
    forbidden: "forbidden",
    idempotency_error: "idempotency_error",
    internal_server_error: "internal_server_error",
    invalid_request: "invalid_request",
    invalid_sql_query: "invalid_sql_query",
    invalid_signature: "invalid_signature",
    malformed_transaction: "malformed_transaction",
    not_found: "not_found",
    payment_method_required: "payment_method_required",
    payment_required: "payment_required",
    settlement_failed: "settlement_failed",
    rate_limit_exceeded: "rate_limit_exceeded",
    request_canceled: "request_canceled",
    service_unavailable: "service_unavailable",
    timed_out: "timed_out",
    unauthorized: "unauthorized",
    policy_violation: "policy_violation",
    policy_in_use: "policy_in_use",
    account_limit_exceeded: "account_limit_exceeded",
    network_not_tradable: "network_not_tradable",
    guest_permission_denied: "guest_permission_denied",
    guest_region_forbidden: "guest_region_forbidden",
    guest_transaction_limit: "guest_transaction_limit",
    guest_transaction_count: "guest_transaction_count",
    phone_number_verification_expired: "phone_number_verification_expired",
    document_verification_failed: "document_verification_failed",
    recipient_allowlist_violation: "recipient_allowlist_violation",
    recipient_allowlist_pending: "recipient_allowlist_pending",
    travel_rules_recipient_violation: "travel_rules_recipient_violation",
    source_account_invalid: "source_account_invalid",
    target_account_invalid: "target_account_invalid",
    source_account_not_found: "source_account_not_found",
    target_account_not_found: "target_account_not_found",
    source_asset_not_supported: "source_asset_not_supported",
    target_asset_not_supported: "target_asset_not_supported",
    target_email_invalid: "target_email_invalid",
    target_onchain_address_invalid: "target_onchain_address_invalid",
    transfer_amount_invalid: "transfer_amount_invalid",
    transfer_asset_not_supported: "transfer_asset_not_supported",
    insufficient_balance: "insufficient_balance",
    metadata_too_many_entries: "metadata_too_many_entries",
    metadata_key_too_long: "metadata_key_too_long",
    metadata_value_too_long: "metadata_value_too_long",
    travel_rules_field_missing: "travel_rules_field_missing",
    asset_mismatch: "asset_mismatch",
    mfa_already_enrolled: "mfa_already_enrolled",
    mfa_invalid_code: "mfa_invalid_code",
    mfa_flow_expired: "mfa_flow_expired",
    mfa_required: "mfa_required",
    mfa_not_enrolled: "mfa_not_enrolled",
    order_quote_expired: "order_quote_expired",
    order_already_filled: "order_already_filled",
    order_already_canceled: "order_already_canceled",
    account_not_ready: "account_not_ready",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmEip7702DelegationNetwork = {
    "base-sepolia": "base-sepolia",
    base: "base",
    arbitrum: "arbitrum",
    optimism: "optimism",
    polygon: "polygon",
    ethereum: "ethereum",
    "ethereum-sepolia": "ethereum-sepolia",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmEip7702DelegationOperationStatus = {
    UNSPECIFIED: "UNSPECIFIED",
    PENDING: "PENDING",
    SUBMITTED: "SUBMITTED",
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmUserOperationNetwork = {
    "base-sepolia": "base-sepolia",
    base: "base",
    arbitrum: "arbitrum",
    optimism: "optimism",
    zora: "zora",
    polygon: "polygon",
    bnb: "bnb",
    avalanche: "avalanche",
    ethereum: "ethereum",
    "ethereum-sepolia": "ethereum-sepolia",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmUserOperationStatus = {
    pending: "pending",
    signed: "signed",
    broadcast: "broadcast",
    complete: "complete",
    dropped: "dropped",
    failed: "failed",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SpendPermissionNetwork = {
    base: "base",
    "base-sepolia": "base-sepolia",
    ethereum: "ethereum",
    "ethereum-sepolia": "ethereum-sepolia",
    optimism: "optimism",
    arbitrum: "arbitrum",
    avalanche: "avalanche",
    polygon: "polygon",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmSwapsNetwork = {
    base: "base",
    ethereum: "ethereum",
    arbitrum: "arbitrum",
    optimism: "optimism",
    polygon: "polygon",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ListEvmTokenBalancesNetwork = {
    base: "base",
    "base-sepolia": "base-sepolia",
    ethereum: "ethereum",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EthValueCriterionType = {
    ethValue: "ethValue",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EthValueCriterionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmAddressCriterionType = {
    evmAddress: "evmAddress",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmAddressCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.KnownAbiType = {
    erc20: "erc20",
    erc721: "erc721",
    erc1155: "erc1155",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.AbiStateMutability = {
    pure: "pure",
    view: "view",
    nonpayable: "nonpayable",
    payable: "payable",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.AbiFunctionType = {
    function: "function",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.AbiInputType = {
    constructor: "constructor",
    error: "error",
    event: "event",
    fallback: "fallback",
    receive: "receive",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmDataParameterConditionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmDataParameterConditionListOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmDataCriterionType = {
    evmData: "evmData",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.NetUSDChangeCriterionType = {
    netUSDChange: "netUSDChange",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.NetUSDChangeCriterionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTransactionRuleOperation = {
    signEvmTransaction: "signEvmTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmNetworkCriterionType = {
    evmNetwork: "evmNetwork",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmNetworkCriterionNetworksItem = {
    "base-sepolia": "base-sepolia",
    base: "base",
    ethereum: "ethereum",
    "ethereum-sepolia": "ethereum-sepolia",
    avalanche: "avalanche",
    polygon: "polygon",
    optimism: "optimism",
    arbitrum: "arbitrum",
    zora: "zora",
    bnb: "bnb",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmNetworkCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEvmTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEvmTransactionRuleOperation = {
    sendEvmTransaction: "sendEvmTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmMessageCriterionType = {
    evmMessage: "evmMessage",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmMessageRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmMessageRuleOperation = {
    signEvmMessage: "signEvmMessage",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmTypedAddressConditionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.EvmTypedNumericalConditionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTypedDataFieldCriterionType = {
    evmTypedDataField: "evmTypedDataField",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTypedDataVerifyingContractCriterionType = {
    evmTypedDataVerifyingContract: "evmTypedDataVerifyingContract",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTypedDataVerifyingContractCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTypedDataRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmTypedDataRuleOperation = {
    signEvmTypedData: "signEvmTypedData",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolAddressCriterionType = {
    solAddress: "solAddress",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolAddressCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolValueCriterionType = {
    solValue: "solValue",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolValueCriterionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SplAddressCriterionType = {
    splAddress: "splAddress",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SplAddressCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SplValueCriterionType = {
    splValue: "splValue",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SplValueCriterionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.MintAddressCriterionType = {
    mintAddress: "mintAddress",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.MintAddressCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.KnownIdlType = {
    SystemProgram: "SystemProgram",
    TokenProgram: "TokenProgram",
    AssociatedTokenProgram: "AssociatedTokenProgram",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolDataParameterConditionOperator = {
    ">": ">",
    ">=": ">=",
    "<": "<",
    "<=": "<=",
    "==": "==",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolDataParameterConditionListOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolDataCriterionType = {
    solData: "solData",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ProgramIdCriterionType = {
    programId: "programId",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ProgramIdCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignSolTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignSolTransactionRuleOperation = {
    signSolTransaction: "signSolTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolNetworkCriterionType = {
    solNetwork: "solNetwork",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolNetworkCriterionNetworksItem = {
    "solana-devnet": "solana-devnet",
    solana: "solana",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolNetworkCriterionOperator = {
    in: "in",
    not_in: "not in",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendSolTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendSolTransactionRuleOperation = {
    sendSolTransaction: "sendSolTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SolMessageCriterionType = {
    solMessage: "solMessage",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignSolMessageRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignSolMessageRuleOperation = {
    signSolMessage: "signSolMessage",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmHashRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEvmHashRuleOperation = {
    signEvmHash: "signEvmHash",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.PrepareUserOperationRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.PrepareUserOperationRuleOperation = {
    prepareUserOperation: "prepareUserOperation",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendUserOperationRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendUserOperationRuleOperation = {
    sendUserOperation: "sendUserOperation",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserEvmTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserEvmTransactionRuleOperation = {
    signEndUserEvmTransaction: "signEndUserEvmTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEndUserEvmTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEndUserEvmTransactionRuleOperation = {
    sendEndUserEvmTransaction: "sendEndUserEvmTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserEvmMessageRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserEvmMessageRuleOperation = {
    signEndUserEvmMessage: "signEndUserEvmMessage",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserEvmTypedDataRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserEvmTypedDataRuleOperation = {
    signEndUserEvmTypedData: "signEndUserEvmTypedData",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserSolTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserSolTransactionRuleOperation = {
    signEndUserSolTransaction: "signEndUserSolTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEndUserSolTransactionRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEndUserSolTransactionRuleOperation = {
    sendEndUserSolTransaction: "sendEndUserSolTransaction",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserSolMessageRuleAction = {
    reject: "reject",
    accept: "accept",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SignEndUserSolMessageRuleOperation = {
    signEndUserSolMessage: "signEndUserSolMessage",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.PolicyScope = {
    project: "project",
    account: "account",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ListSolanaTokenBalancesNetwork = {
    solana: "solana",
    "solana-devnet": "solana-devnet",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnchainDataResultSchemaColumnsItemType = {
    String: "String",
    UInt8: "UInt8",
    UInt16: "UInt16",
    UInt32: "UInt32",
    UInt64: "UInt64",
    UInt128: "UInt128",
    UInt256: "UInt256",
    Int8: "Int8",
    Int16: "Int16",
    Int32: "Int32",
    Int64: "Int64",
    Int128: "Int128",
    Int256: "Int256",
    Float32: "Float32",
    Float64: "Float64",
    Bool: "Bool",
    Date: "Date",
    DateTime: "DateTime",
    DateTime64: "DateTime64",
    UUID: "UUID",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402Version = {
    NUMBER_1: 1,
    NUMBER_2: 2,
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402V1PaymentPayloadScheme = {
    exact: "exact",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402V1PaymentPayloadNetwork = {
    "base-sepolia": "base-sepolia",
    base: "base",
    "solana-devnet": "solana-devnet",
    solana: "solana",
    polygon: "polygon",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402V2PaymentRequirementsScheme = {
    exact: "exact",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402V1PaymentRequirementsScheme = {
    exact: "exact",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402V1PaymentRequirementsNetwork = {
    "base-sepolia": "base-sepolia",
    base: "base",
    "solana-devnet": "solana-devnet",
    solana: "solana",
    polygon: "polygon",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402VerifyInvalidReason = {
    insufficient_funds: "insufficient_funds",
    invalid_scheme: "invalid_scheme",
    invalid_network: "invalid_network",
    invalid_x402_version: "invalid_x402_version",
    invalid_payment_requirements: "invalid_payment_requirements",
    invalid_payload: "invalid_payload",
    invalid_exact_evm_payload_authorization_value: "invalid_exact_evm_payload_authorization_value",
    invalid_exact_evm_payload_authorization_value_too_low: "invalid_exact_evm_payload_authorization_value_too_low",
    invalid_exact_evm_payload_authorization_valid_after: "invalid_exact_evm_payload_authorization_valid_after",
    invalid_exact_evm_payload_authorization_valid_before: "invalid_exact_evm_payload_authorization_valid_before",
    invalid_exact_evm_payload_authorization_typed_data_message: "invalid_exact_evm_payload_authorization_typed_data_message",
    invalid_exact_evm_payload_authorization_from_address_kyt: "invalid_exact_evm_payload_authorization_from_address_kyt",
    invalid_exact_evm_payload_authorization_to_address_kyt: "invalid_exact_evm_payload_authorization_to_address_kyt",
    invalid_exact_evm_payload_signature: "invalid_exact_evm_payload_signature",
    invalid_exact_evm_payload_signature_address: "invalid_exact_evm_payload_signature_address",
    invalid_exact_evm_permit2_payload_allowance_required: "invalid_exact_evm_permit2_payload_allowance_required",
    invalid_exact_evm_permit2_payload_signature: "invalid_exact_evm_permit2_payload_signature",
    invalid_exact_evm_permit2_payload_deadline: "invalid_exact_evm_permit2_payload_deadline",
    invalid_exact_evm_permit2_payload_valid_after: "invalid_exact_evm_permit2_payload_valid_after",
    invalid_exact_evm_permit2_payload_spender: "invalid_exact_evm_permit2_payload_spender",
    invalid_exact_evm_permit2_payload_recipient: "invalid_exact_evm_permit2_payload_recipient",
    invalid_exact_evm_permit2_payload_amount: "invalid_exact_evm_permit2_payload_amount",
    invalid_exact_svm_payload_transaction: "invalid_exact_svm_payload_transaction",
    invalid_exact_svm_payload_transaction_amount_mismatch: "invalid_exact_svm_payload_transaction_amount_mismatch",
    invalid_exact_svm_payload_transaction_create_ata_instruction: "invalid_exact_svm_payload_transaction_create_ata_instruction",
    invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_payee: "invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_payee",
    invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_asset: "invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_asset",
    invalid_exact_svm_payload_transaction_instructions: "invalid_exact_svm_payload_transaction_instructions",
    invalid_exact_svm_payload_transaction_instructions_length: "invalid_exact_svm_payload_transaction_instructions_length",
    invalid_exact_svm_payload_transaction_instructions_compute_limit_instruction: "invalid_exact_svm_payload_transaction_instructions_compute_limit_instruction",
    invalid_exact_svm_payload_transaction_instructions_compute_price_instruction: "invalid_exact_svm_payload_transaction_instructions_compute_price_instruction",
    invalid_exact_svm_payload_transaction_instructions_compute_price_instruction_too_high: "invalid_exact_svm_payload_transaction_instructions_compute_price_instruction_too_high",
    invalid_exact_svm_payload_transaction_instruction_not_spl_token_transfer_checked: "invalid_exact_svm_payload_transaction_instruction_not_spl_token_transfer_checked",
    invalid_exact_svm_payload_transaction_instruction_not_token_2022_transfer_checked: "invalid_exact_svm_payload_transaction_instruction_not_token_2022_transfer_checked",
    invalid_exact_svm_payload_transaction_not_a_transfer_instruction: "invalid_exact_svm_payload_transaction_not_a_transfer_instruction",
    invalid_exact_svm_payload_transaction_cannot_derive_receiver_ata: "invalid_exact_svm_payload_transaction_cannot_derive_receiver_ata",
    invalid_exact_svm_payload_transaction_receiver_ata_not_found: "invalid_exact_svm_payload_transaction_receiver_ata_not_found",
    invalid_exact_svm_payload_transaction_sender_ata_not_found: "invalid_exact_svm_payload_transaction_sender_ata_not_found",
    invalid_exact_svm_payload_transaction_simulation_failed: "invalid_exact_svm_payload_transaction_simulation_failed",
    invalid_exact_svm_payload_transaction_transfer_to_incorrect_ata: "invalid_exact_svm_payload_transaction_transfer_to_incorrect_ata",
    invalid_exact_svm_payload_transaction_fee_payer_included_in_instruction_accounts: "invalid_exact_svm_payload_transaction_fee_payer_included_in_instruction_accounts",
    invalid_exact_svm_payload_transaction_fee_payer_transferring_funds: "invalid_exact_svm_payload_transaction_fee_payer_transferring_funds",
    unknown_error: "unknown_error",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402SettleErrorReason = {
    insufficient_funds: "insufficient_funds",
    invalid_scheme: "invalid_scheme",
    invalid_network: "invalid_network",
    invalid_x402_version: "invalid_x402_version",
    invalid_payment_requirements: "invalid_payment_requirements",
    invalid_payload: "invalid_payload",
    invalid_exact_evm_payload_authorization_value: "invalid_exact_evm_payload_authorization_value",
    invalid_exact_evm_payload_authorization_value_too_low: "invalid_exact_evm_payload_authorization_value_too_low",
    invalid_exact_evm_payload_authorization_valid_after: "invalid_exact_evm_payload_authorization_valid_after",
    invalid_exact_evm_payload_authorization_valid_before: "invalid_exact_evm_payload_authorization_valid_before",
    invalid_exact_evm_payload_authorization_typed_data_message: "invalid_exact_evm_payload_authorization_typed_data_message",
    invalid_exact_evm_payload_authorization_from_address_kyt: "invalid_exact_evm_payload_authorization_from_address_kyt",
    invalid_exact_evm_payload_authorization_to_address_kyt: "invalid_exact_evm_payload_authorization_to_address_kyt",
    invalid_exact_evm_payload_signature: "invalid_exact_evm_payload_signature",
    invalid_exact_evm_payload_signature_address: "invalid_exact_evm_payload_signature_address",
    invalid_exact_evm_permit2_payload_allowance_required: "invalid_exact_evm_permit2_payload_allowance_required",
    invalid_exact_evm_permit2_payload_signature: "invalid_exact_evm_permit2_payload_signature",
    invalid_exact_evm_permit2_payload_deadline: "invalid_exact_evm_permit2_payload_deadline",
    invalid_exact_evm_permit2_payload_valid_after: "invalid_exact_evm_permit2_payload_valid_after",
    invalid_exact_evm_permit2_payload_spender: "invalid_exact_evm_permit2_payload_spender",
    invalid_exact_evm_permit2_payload_recipient: "invalid_exact_evm_permit2_payload_recipient",
    invalid_exact_evm_permit2_payload_amount: "invalid_exact_evm_permit2_payload_amount",
    invalid_exact_svm_payload_transaction: "invalid_exact_svm_payload_transaction",
    invalid_exact_svm_payload_transaction_amount_mismatch: "invalid_exact_svm_payload_transaction_amount_mismatch",
    invalid_exact_svm_payload_transaction_create_ata_instruction: "invalid_exact_svm_payload_transaction_create_ata_instruction",
    invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_payee: "invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_payee",
    invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_asset: "invalid_exact_svm_payload_transaction_create_ata_instruction_incorrect_asset",
    invalid_exact_svm_payload_transaction_instructions: "invalid_exact_svm_payload_transaction_instructions",
    invalid_exact_svm_payload_transaction_instructions_length: "invalid_exact_svm_payload_transaction_instructions_length",
    invalid_exact_svm_payload_transaction_instructions_compute_limit_instruction: "invalid_exact_svm_payload_transaction_instructions_compute_limit_instruction",
    invalid_exact_svm_payload_transaction_instructions_compute_price_instruction: "invalid_exact_svm_payload_transaction_instructions_compute_price_instruction",
    invalid_exact_svm_payload_transaction_instructions_compute_price_instruction_too_high: "invalid_exact_svm_payload_transaction_instructions_compute_price_instruction_too_high",
    invalid_exact_svm_payload_transaction_instruction_not_spl_token_transfer_checked: "invalid_exact_svm_payload_transaction_instruction_not_spl_token_transfer_checked",
    invalid_exact_svm_payload_transaction_instruction_not_token_2022_transfer_checked: "invalid_exact_svm_payload_transaction_instruction_not_token_2022_transfer_checked",
    invalid_exact_svm_payload_transaction_not_a_transfer_instruction: "invalid_exact_svm_payload_transaction_not_a_transfer_instruction",
    invalid_exact_svm_payload_transaction_cannot_derive_receiver_ata: "invalid_exact_svm_payload_transaction_cannot_derive_receiver_ata",
    invalid_exact_svm_payload_transaction_receiver_ata_not_found: "invalid_exact_svm_payload_transaction_receiver_ata_not_found",
    invalid_exact_svm_payload_transaction_sender_ata_not_found: "invalid_exact_svm_payload_transaction_sender_ata_not_found",
    invalid_exact_svm_payload_transaction_simulation_failed: "invalid_exact_svm_payload_transaction_simulation_failed",
    invalid_exact_svm_payload_transaction_transfer_to_incorrect_ata: "invalid_exact_svm_payload_transaction_transfer_to_incorrect_ata",
    invalid_exact_svm_payload_transaction_fee_payer_included_in_instruction_accounts: "invalid_exact_svm_payload_transaction_fee_payer_included_in_instruction_accounts",
    invalid_exact_svm_payload_transaction_fee_payer_transferring_funds: "invalid_exact_svm_payload_transaction_fee_payer_transferring_funds",
    settle_exact_evm_transaction_confirmation_timed_out: "settle_exact_evm_transaction_confirmation_timed_out",
    settle_exact_node_failure: "settle_exact_node_failure",
    settle_exact_failed_onchain: "settle_exact_failed_onchain",
    settle_exact_svm_block_height_exceeded: "settle_exact_svm_block_height_exceeded",
    settle_exact_svm_transaction_confirmation_timed_out: "settle_exact_svm_transaction_confirmation_timed_out",
    unknown_error: "unknown_error",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402SupportedPaymentKindScheme = {
    exact: "exact",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.X402SupportedPaymentKindNetwork = {
    "base-sepolia": "base-sepolia",
    base: "base",
    "solana-devnet": "solana-devnet",
    solana: "solana",
    polygon: "polygon",
    "eip155:8453": "eip155:8453",
    "eip155:84532": "eip155:84532",
    "eip155:137": "eip155:137",
    "solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp": "solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp",
    "solana:EtWTRABZaYq6iMfeYKouRu166VU2xqa1": "solana:EtWTRABZaYq6iMfeYKouRu166VU2xqa1",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampOrderPaymentMethodTypeId = {
    GUEST_CHECKOUT_APPLE_PAY: "GUEST_CHECKOUT_APPLE_PAY",
    GUEST_CHECKOUT_GOOGLE_PAY: "GUEST_CHECKOUT_GOOGLE_PAY",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampOrderFeeType = {
    FEE_TYPE_NETWORK: "FEE_TYPE_NETWORK",
    FEE_TYPE_EXCHANGE: "FEE_TYPE_EXCHANGE",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampOrderStatus = {
    ONRAMP_ORDER_STATUS_PENDING_AUTH: "ONRAMP_ORDER_STATUS_PENDING_AUTH",
    ONRAMP_ORDER_STATUS_PENDING_PAYMENT: "ONRAMP_ORDER_STATUS_PENDING_PAYMENT",
    ONRAMP_ORDER_STATUS_PROCESSING: "ONRAMP_ORDER_STATUS_PROCESSING",
    ONRAMP_ORDER_STATUS_COMPLETED: "ONRAMP_ORDER_STATUS_COMPLETED",
    ONRAMP_ORDER_STATUS_FAILED: "ONRAMP_ORDER_STATUS_FAILED",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampPaymentLinkType = {
    PAYMENT_LINK_TYPE_APPLE_PAY_BUTTON: "PAYMENT_LINK_TYPE_APPLE_PAY_BUTTON",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampQuotePaymentMethodTypeId = {
    CARD: "CARD",
    ACH: "ACH",
    APPLE_PAY: "APPLE_PAY",
    PAYPAL: "PAYPAL",
    FIAT_WALLET: "FIAT_WALLET",
    CRYPTO_WALLET: "CRYPTO_WALLET",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampUserIdType = {
    phone_number: "phone_number",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.OnrampLimitType = {
    weekly_spending: "weekly_spending",
    lifetime_transactions: "lifetime_transactions",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ListEndUsersSortItem = {
    "createdAt=asc": "createdAt=asc",
    "createdAt=desc": "createdAt=desc",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ImportEndUserBodyKeyType = {
    evm: "evm",
    solana: "solana",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendEvmTransactionBodyNetwork = {
    base: "base",
    "base-sepolia": "base-sepolia",
    ethereum: "ethereum",
    "ethereum-sepolia": "ethereum-sepolia",
    avalanche: "avalanche",
    polygon: "polygon",
    optimism: "optimism",
    arbitrum: "arbitrum",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.RequestEvmFaucetBodyNetwork = {
    "base-sepolia": "base-sepolia",
    "ethereum-sepolia": "ethereum-sepolia",
    "ethereum-hoodi": "ethereum-hoodi",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.RequestEvmFaucetBodyToken = {
    eth: "eth",
    usdc: "usdc",
    eurc: "eurc",
    cbbtc: "cbbtc",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.ListPoliciesScope = {
    project: "project",
    account: "account",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.CreatePolicyBodyScope = {
    project: "project",
    account: "account",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.SendSolanaTransactionBodyNetwork = {
    solana: "solana",
    "solana-devnet": "solana-devnet",
};
// eslint-disable-next-line @typescript-eslint/no-redeclare
exports.RequestSolanaFaucetBodyToken = {
    sol: "sol",
    usdc: "usdc",
};
